pkg/cache/server/server.go

```diff
@@ -37,11 +37,11 @@ func NewServer(c CompletedConfig) (*Server, error) {
 	return s, nil
 }
 
-func (s *Server) Run(ctx context.Context) error {
+func (s *Server) Complete(ctx context.Context) (*genericapiserver.GenericAPIServer, error) {
```

Comment:
why? This is not completion semantics anymore.

Reply:
I need this intermediary step when running the cache-server with the kcp. We can change the name. 

Basically, I will:
```
newCacheServer, err := cacheserver.NewServer(completedCacheServerConfig)
if err != nil {
	return err
}
completedCacheServer, err := newCacheServer.Complete(ctx)
if err != nil {
	return err
}

preparedCacheServer := completedCacheServer.PrepareRun()
if err := s.AddPostStartHook("kcp-start-cache-server", func(ctx genericapiserver.PostStartHookContext) error {
	preparedCacheServer.RunPostStartHooks(ctx.StopCh)
	return nil
}); err != nil {
	return err
}

s.preHandlerChainMux.Handle(virtualcommandoptions.DefaultRootPathPrefix+"/cache/", preparedCacheServer.Handler)
s.preHandlerChainMux.Handle("/shards/", preparedCacheServer.Handler)

```